### PR TITLE
bug: Remove execution context determination

### PR DESCRIPTION
**Pull request by mikechu-optimizely**
## Summary
- Removing an attempt to distinguish between execution contexts (browser vs. node).  
- We found the in demo prep, this implementation was incorrect and causing bugs
- We need a better solution to handle context determination
- For now, simple check for the presence of `window` === Browser and `process` === Node is being used for the new code path
- This PR relates to [786](https://github.com/optimizely/javascript-sdk/pull/786)
 
## Test plan
- All tests still pass after removal

## Issues
- FSSDK-8386 (OASIS-8386 [to pass jira_ticket_reference_check])

### CODE REVIEW

File: packages/optimizely-sdk/lib/index.browser.ts [hunk 1]
```diff
@@ -31,9 +31,7 @@ import eventProcessorConfigValidator from './utils/event_processor_config_valida
 import { createNotificationCenter } from './core/notification_center';
 import { default as eventProcessor } from './plugins/event_processor';
 import { OptimizelyDecideOption, Client, Config } from './shared_types';
-import { createHttpPollingDatafileManager } from './plugins/datafile_manager/browser_http_polling_datafile_manager';
-import { EXECUTION_CONTEXT_TYPE } from './utils/enums';
-import { ExecutionContext } from './utils/execution_context';
+import { createHttpPollingDatafileManager } from './plugins/datafile_manager/http_polling_datafile_manager';
```
Comment (mikechu-optimizely) [inline packages/optimizely-sdk/lib/index.browser.ts:34]: @zashraf1985 @ozayr-zaviar I'm worried that I may not have merged this correctly.

File: packages/optimizely-sdk/lib/index.react_native.ts [hunk 2]
```diff
@@ -31,9 +31,7 @@ import eventProcessorConfigValidator from './utils/event_processor_config_valida
 import { createNotificationCenter } from './core/notification_center';
 import { createEventProcessor } from './plugins/event_processor/index.react_native';
 import { OptimizelyDecideOption, Client, Config } from './shared_types';
-import { createHttpPollingDatafileManager } from './plugins/datafile_manager/react_native_http_polling_datafile_manager';
-import { EXECUTION_CONTEXT_TYPE } from './utils/enums';
-import { ExecutionContext } from './utils/execution_context';
+import { createHttpPollingDatafileManager } from './plugins/datafile_manager/http_polling_datafile_manager';
```
Comment (mikechu-optimizely) [inline packages/optimizely-sdk/lib/index.react_native.ts:34]: This too might not be correct either.  

Should it be `./plugins/datafile_manager/react_native_http_polling_datafile_manager` or `'./plugins/datafile_manager/http_polling_datafile_manager'`?

@zashraf1985 @ozayr-zaviar

File: packages/optimizely-sdk/lib/index.browser.ts (hunk 1 above)
Comment (mikechu-optimizely) [reply packages/optimizely-sdk/lib/index.browser.ts:34]: I think this should probably be `'./plugins/datafile_manager/browser_http_polling_datafile_manager'` I think

File: packages/optimizely-sdk/lib/index.react_native.ts (hunk 2 above)
Comment (mikechu-optimizely) [reply packages/optimizely-sdk/lib/index.react_native.ts:34]: I think this one should be `react_native_http_polling_datafile_manager`.  Grrr. Sorry I messed this merge up.

File: packages/optimizely-sdk/lib/utils/http_request_handler/request_handler_factory.ts [hunk 3]
```diff
@@ -18,21 +18,18 @@ import { LogHandler } from '../../modules/logging';
...
-      case EXECUTION_CONTEXT_TYPE.NODE:
-        return new NodeRequestHandler(logger, timeout);
-      default:
-        return null as unknown as RequestHandler;
+    if (window) {
+      return new BrowserRequestHandler(logger, timeout);
+    } else if (process) {
+      return new NodeRequestHandler(logger, timeout);
+    } else {
+      return null as unknown as RequestHandler;
```
Comment (mikechu-optimizely) [inline packages/optimizely-sdk/lib/utils/http_request_handler/request_handler_factory.ts:32]: This will get removed by #786

Comment (mikechu-optimizely) [reply packages/optimizely-sdk/lib/utils/http_request_handler/request_handler_factory.ts:32]: Well the whole file will...

**Review by zashraf1985 — APPROVED** [review]

**Pull request merged**